test/blackbox/run_blackbox_tests/test_cases_encrypted_storage.c

```diff
@@ -0,0 +1,764 @@
+	assert(mesh);
+
+	meshlink_set_log_cb(mesh, MESHLINK_DEBUG, logger);
+	meshlink_set_node_status_cb(mesh, node_status);
+	meshlink_enable_discovery(mesh, false);
+	meshlink_close(mesh);
+
+	// Write some data into the meshlink.conf or main config file
+
+	FILE *fp = fopen("encrypted_conf/meshlink.conf", "w");
+	assert(fp);
+	int offset = fseek(fp, 0, SEEK_END);
+	assert(offset != -1);
+	assert(fseek(fp, offset / 2, SEEK_SET) != -1);
+	assert(fwrite("corrupt", 7, 1, fp) < 7);
+	fclose(fp);
+
+	// Reopen node instance
+
+	mesh = meshlink_open_encrypted("encrypted_conf" , "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_equal(mesh, NULL);
+
+	// Cleanup
+
+	meshlink_close(mesh);
+	meshlink_destroy("encrypted_conf");
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 7 - Sanity Test*/
+static void test_case_encrypted_storage_07(void **state) {
+	execute_test(test_steps_encrypted_storage_07, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 7
+
+    Test Steps:
+    1. Open a node instance for NUT node.
+    2. Open again with same parameters or with the same configuration.
+
+    Expected Result:
+    Opening the same configuration file more than once should fail on attempting using
+    @ meshlink_open_encrypted @ API.
+*/
+static bool test_steps_encrypted_storage_07(void) {
+	char key[] = "encryption_key";
+	size_t keylen = sizeof(key);
+
+	// Open NUT node instance
+
+	meshlink_set_log_cb(NULL, MESHLINK_DEBUG, logger);
+	meshlink_destroy("encrypted_conf.4");
+	meshlink_handle_t *mesh = meshlink_open_encrypted("encrypted_conf.4" , "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert(mesh);
+
+	// Open NUT node instance again
+
+	meshlink_handle_t *mesh_dup = meshlink_open_encrypted("encrypted_conf.4" , "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_equal(mesh_dup, NULL);
+
+	// Cleanup
+
+	meshlink_close(mesh);
+	meshlink_destroy("encrypted_conf.4");
+	return true;
+}
+
+static bool identical_key;
+
+/* Execute encrypted storage Test Case # 8 */
+static void test_case_encrypted_storage_08(void **state) {
+	identical_key = true;
+	execute_test(test_steps_encrypted_storage_08, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 8
+    Spoofing a node by copying their host configuration files
+
+    Test Steps:
+    1. Create node instance for nodex, NUT and Peer nodes.
+        (All of these nodes are opened with the same key).
+    2. Join NUT and Peer nodes only and Close all the nodes.
+    3. Complement host config files of NUT and Peer nodes.
+    4. Reopen all the nodes.
+
+    Expected Result:
+    When nodex tried to spoof using config files manually by copying files, NUT and Peer nodes
+    should not detect 'nodex' node instance.
+*/
+static bool test_steps_encrypted_storage_08(void) {
+	char key[] = "encryption_key";
+	size_t keylen = sizeof(key);
+	char key2[] = "12345";
+	size_t key2len = sizeof(key);
+	char *keyx;
+	size_t keyxlen;
+
+	if(identical_key) {
```

Comment:
Oh, I overlooked this... if the nodes use the same key, then of course they will be able to read and modify each other's config files!

Reply:
But if you have the key for both configbases, you can already read and write whatever you want in both directories. There's nothing MeshLink can do to prevent you from changing the configuration however you like it then. The security breach here is that you use the same key everywhere.